### ISSUE TITLE
ERROR in Template execution failed: ReferenceError: htmlWebpackPlugin is not defined

### ISSUE THREAD

**Issue by unbugx** [ampedandwired/html-webpack-plugin#323]
Hi there! I updated to latest version 2.16.2 and now have this problem:
 
ERROR in Template execution failed: ReferenceError: htmlWebpackPlugin is not defined

ERROR in   ReferenceError: htmlWebpackPlugin is not defined
  
  - index.js:246 
    [static-analysis]/[html-webpack-plugin]/index.js:246:16
  
  - util.js:16 tryCatcher
    [static-analysis]/[html-webpack-plugin]/[bluebird]/js/release/util.js:16:23
  
  - promise.js:502 Promise._settlePromiseFromHandler
    [static-analysis]/[html-webpack-plugin]/[bluebird]/js/release/promise.js:502:31
  
  - promise.js:559 Promise._settlePromise
    [static-analysis]/[html-webpack-plugin]/[bluebird]/js/release/promise.js:559:18
  
  - promise.js:596 Promise._settlePromiseCtx
    [static-analysis]/[html-webpack-plugin]/[bluebird]/js/release/promise.js:596:10
  
  - async.js:143 Async._drainQueue
    [static-analysis]/[html-webpack-plugin]/[bluebird]/js/release/async.js:143:12
  
  - async.js:148 Async._drainQueues
    [static-analysis]/[html-webpack-plugin]/[bluebird]/js/release/async.js:148:10
  
  - async.js:17 Immediate.Async.drainQueues [as _onImmediate]
    [static-analysis]/[html-webpack-plugin]/[bluebird]/js/release/async.js:17:14

**Comment by unbugx** [comment]
It's magic, now it works.

**Comment by jantimon** [comment]
+1 for magic! ;)

**Comment by alexrqs** [comment]
same issue here, for me was this  : https://github.com/okonet/ejs-loade
```js
{
        test: /\.ejs$/,
        loader: 'ejs-loader?variable=data'
},
```

**Comment by Wizzercn** [comment]
@alexrqs 请问解决了吗，我都折腾一周了也没找到解决办法

**Comment by jozhi** [comment]
following continued ..

**Comment by Wizzercn** [comment]
@jozhi @alexrqs  我解决了..详见:

https://www.wizzer.cn/archives/3293

**Comment by jaxxreal** [comment]
It's not a magic, guys.

For me it happens when I'm trying user ES2015 template literals in html template.

Like this:

`const elem = form.querySelector(`${input[0]}-yui`);`

After it, I'll get this from webpack:
```
ERROR in Template execution failed: ReferenceError: input is not defined

ERROR in   ReferenceError: input is not defined
```

**Comment by jesseminn** [comment]
@jaxxreal I've encountered the same issue when I was trying to use template string. Template string works when without interpolations, but once you interpolate some variable, Webpack reports `ReferenceError`

```js
const ruleSet = document.createTextNode(`
  .${className} {
    width: 100%;
    height: 100vh;
    background-color: black;

    display: flex;
   }
`);
```

**Comment by asumaran** [comment]
This is also happening to me when trying to use JS inside a `<script>` tag on my `index.html` template. using <code>`</code>

**Comment by NicolasRannou** [comment]
Me too - not sure what the problem is but just adding the following code in the html header crashes html webpack plugin:
```
<script>
let name = 'nicolas';
let hello_name = `hello ${name}`;
</script>
```

**Comment by tednyberg** [comment]
Like @NicolasRannou said - string interpolation breaks the plugin, so I'm not sure this issue should be closed? Unless there is another issue for the same/similar problem?

**Comment by balkarov** [comment]
I have the same problem

**Comment by CinKon** [comment]
Happens to me when I want to use `templateParameters`:

```
    new HtmlWebpackPlugin({
      templateParameters: {
        assetsCssImg: utils.kittnConf.dist.cssimg
      },
      filename: 'index.html',
      template: utils.kittnConf.src.structure + 'index.html',
      inject: false,
      hash: false,
      title: utils.kittnConf.template.globalTitle,
      minify: {
        removeComments: true,
        collapseWhitespace: true,
        removeAttributeQuotes: false
      },
      chunksSortMode: 'dependency'
    })
```

```
<!doctype html>
<html lang="de">
<head>
  <meta charset="utf-8"/>

  <% if (htmlWebpackPlugin.options.enviornment != 'development') { %>
  <base href="<?php echo $this->config->server->baseUrl; ?>/" />
  <% } %>
  <title><%= htmlWebpackPlugin.options.title %></title>
  <meta name="twitter:image:src" content="<%= assetsCssImg %>system/twitter-og-image.jpg">
  </script>

</head>

<body>
  <div id="app"></div>

</body>
</html>
```

**Comment by melvik** [comment]
I have the problem in my Vue.js app
        nativeToast({
            message: `New message by ${nickname}`,
            position: 'top',
            timeout: 5000,
            type: 'success'
          })

in Template execution failed: ReferenceError: nickname is not defined

**Comment by jantimon** [comment]
Will be fixed in 4.0.0 alpha: https://github.com/jantimon/html-webpack-plugin/pull/953
